tests/common/mock_cgroup_commands.py

```diff
@@ -68,70 +67,99 @@
+    corresponding path in the matching item. If there is no match, then it checks if the file is under one of the paths
+    in  __MOCKED_PATHS and map the path to the given tmp_dir (e.g. "/lib/systemd/system" becomes
+    "<tmp_dir>/lib/systemd/system".) If there no matches, the path is not changed. Once this mapping has completed
+    the mocks invoke the corresponding original function.
+
+    Matches are done using regular expressions; the regular expressions in __MOCKED_PATHS must create group 0 to indicate
+    the section of the path that needs to be mapped (i.e. use parenthesis around the section that needs to be mapped.)
+    done using
+
+    The command output used in __MOCKED_COMMANDS come from an Ubuntu 18 system.
+    """
     original_popen = subprocess.Popen
-    original_read_file = fileutil.read_file
-    original_write_file = fileutil.write_file
+    original_mkdir = fileutil.mkdir
     original_path_exists = os.path.exists
+    original_open = open
 
-    def add_file(pattern, file_path):
-        patcher.files.insert(0, (pattern, file_path))
-
-    def add_command(pattern, output):
-        patcher.commands.insert(0, (pattern, output))
+    def add_command_mock(pattern, output):
+        patcher.__commands.insert(0, (pattern, output))
 
     def mock_popen(command, *args, **kwargs):
         if isinstance(command, list):
             command_string = " ".join(command)
         else:
             command_string = command
 
-        for cmd in patcher.commands:
+        for cmd in patcher.__commands:
             match = re.match(cmd[0], command_string)
             if match is not None:
                 command = ["echo", cmd[1]]
                 break
 
         return original_popen(command, *args, **kwargs)
-    
-    def mock_read_file(filepath, **kwargs):
-        for item in patcher.files:
-            match = re.match(item[0], filepath)
-            if match is not None:
-                filepath = item[1]
-        return original_read_file(filepath, **kwargs)
 
-    def mock_write_file(filepath, content, **kwargs):
-        for item in patcher.files:
-            match = re.match(item[0], filepath)
+    def get_mapped_path(path):
+        for item in __MOCKED_FILES:
+            match = re.match(item[0], path)
             if match is not None:
-                filepath = item[1]
-        return original_write_file(filepath, content, **kwargs)
+                return item[1]
+
+        for item in __MOCKED_PATHS:
+            mapped = re.sub(item, r"{0}\1".format(tmp_dir), path)
+            if mapped != path:
+                mapped_parent = os.path.split(mapped)[0]
+                if not original_path_exists(mapped_parent):
+                    os.makedirs(mapped_parent)
+                return mapped
+        return path
+
+    def mock_mkdir(path, *args, **kwargs):
+        return original_mkdir(get_mapped_path(path), *args, **kwargs)
+
+    def mock_open(path, *args, **kwargs):
```

Comment:
Instead of mocking individual functions in fileutil, I went ahead and mocked open. The scope of the mock is limited to __MOCKED_FILES and __MOCKED_PATHS so hopefully this global mock will not introduce issues; we can revisit if needed.

The need to do this came because I added a call to fileutil.findstr_in_file in the cgroup configurator andd initially I did not noticed this function too needed to be mocked.